ethportal-api/src/types/content_key.rs

```diff
@@ -1,12 +1,86 @@
-use ethereum_types::H256;
+use anyhow::anyhow;
+use ethereum_types::{H256, U256, U512};
+use quickcheck::{Arbitrary, Gen};
 use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
 use sha2::{Digest as Sha2Digest, Sha256};
+use sha3::{Digest, Keccak256};
 use ssz::{self, Decode, Encode};
 use ssz_derive::{Decode, Encode};
+use ssz_types::{typenum, FixedVector, VariableList};
 use std::fmt;
 
 use trin_utils::bytes::{hex_decode, hex_encode, hex_encode_compact};
 
+/// SSZ encoded overlay content key as bytes
+pub type RawContentKey = Vec<u8>;
+
+/// A content key type whose content id is the inner value. Allows for the construction
+/// of a content key with an arbitary content ID.
+#[derive(Clone, Debug)]
+pub struct IdentityContentKey {
```

Comment:
I don't think we want to support and expose this struct to the library users of `ethportal-api`. If I get this right, `IdentityContentKey` is used only for testing in Trin and it is not part of portal network specs and the global protocol.

As long as `ethportal-api` expose the `OverlayContentKey` trait, any user has all the tools to create its own internal content key structs (for testing and etc).

I would move this type to `trin-types` crate, same for the `RawContentKey` type above.